services/uibackend/main.go

```diff
@@ -0,0 +1,82 @@
+package main
+
+import (
+	"crypto"
+	"crypto/rand"
+	"crypto/rsa"
+	"crypto/sha256"
+	"crypto/x509"
+	"encoding/base64"
+	"encoding/json"
+	"encoding/pem"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"time"
+
+	"github.com/google/uuid"
+	"github.com/gorilla/mux"
+)
+
+type Onboarding struct {
+	Token string `json:"token"`
+}
+
+func parseKey(s []byte) *rsa.PrivateKey {
+
+	Block, _ := pem.Decode(s)
+	k, err := x509.ParsePKCS1PrivateKey(Block.Bytes)
+	if err != nil {
+		panic(err)
+	}
+	return k
+}
+
+func generateToken(w http.ResponseWriter, r *http.Request) {
```

Comment:
Token generation logic be decoupled from http concerns. 
My suggestion is to have a 

`handleOnboardingToken` function to be registered with the endpoint  (in its own file) which then calls an HTTP agnostic generate token function

Reply:
Done